matchconstraint.go

```diff
@@ -0,0 +1,44 @@
+package claircore
+
+// MatchConstraint explains to the caller how a search for a package's vulnerability should
+// be constrained.
+//
+// for example if sql implementation encounters a DistributionDID constraint
+// it should create a query similar to "SELECT * FROM vulnerabilities WHERE package_name = ? AND distribution_did = ?"
+type MatchConstraint int
+
+//go:generate go run golang.org/x/tools/cmd/stringer -type MatchConstraint
+
+const (
```

Comment:
TODO: If these Constraints stay here, they need more specificity in their naming. `claircore.PackageName` has no indication that it's a contraint.